src/uu/ls/src/ls.rs

```diff
@@ -1671,85 +1781,142 @@ fn display_grid(
-            // GNU `ls` uses a "." character to indicate a file with a security context,
-            // but not other alternate access method.
-            "."
-        } else {
-            ""
-        },
-        pad_left(&display_symlink_count(md), padding.longest_link_count_len),
-    );
 
-    if config.long.owner {
         let _ = write!(
             out,
-            " {}",
-            pad_right(&display_uname(md, config), padding.longest_uname_len)
+            "{}{} {}",
+            display_permissions(&md, true),
+            if item.security_context.len() > 1 {
+                // GNU `ls` uses a "." character to indicate a file with a security context,
+                // but not other alternate access method.
+                "."
+            } else {
+                ""
+            },
+            pad_left(&display_symlink_count(&md), padding.longest_link_count_len),
         );
-    }
 
-    if config.long.group {
-        let _ = write!(
+        if config.long.owner {
+            let _ = write!(
+                out,
+                " {}",
+                pad_right(&display_uname(&md, config), padding.longest_uname_len),
+            );
+        }
+
+        if config.long.group {
+            let _ = write!(
+                out,
+                " {}",
+                pad_right(&display_group(&md, config), padding.longest_group_len),
+            );
+        }
+
+        if config.context {
+            let _ = write!(
+                out,
+                " {}",
+                pad_right(&item.security_context, padding.longest_context_len),
+            );
+        }
+
+        // Author is only different from owner on GNU/Hurd, so we reuse
+        // the owner, since GNU/Hurd is not currently supported by Rust.
+        if config.long.author {
+            let _ = write!(
+                out,
+                " {}",
+                pad_right(&display_uname(&md, config), padding.longest_uname_len),
+            );
+        }
+
+        let dfn = display_file_name(item, config, None, out).contents;
+
+        let _ = writeln!(
             out,
-            " {}",
-            pad_right(&display_group(md, config), padding.longest_group_len)
+            " {} {} {}",
+            pad_left(&display_size_or_rdev(&md, config), padding.longest_size_len),
+            display_date(&md, config),
+            dfn,
         );
-    }
+    } else {
+        // this is expressly for the case of a dangling symlink
+        let _ = out.flush();
+        show!(LsError::IOErrorContext(
+            std::io::Error::new(ErrorKind::NotFound, "NotFound"),
+            item.p_buf.as_os_str().to_owned()
+        ));
+        #[cfg(unix)]
+        {
+            if config.inode {
+                if config.recursive {
+                    let _ = write!(out, "{} ", "?".to_string());
+                } else {
+                    let _ = write!(out, "       {} ", "?".to_string());
+                }
+            }
+        }
```

Comment:
Please leave this out for now (since it conflicts with the other PR and does not provide a full solution)

Reply:
I'm not sure about this one.  I think combined with the changes made at fn display_file_name, it might?  Yes, devices needs better padding?  But what's missing?

Reply:
Hmm, I assumed that this was part of the other inode changes, but maybe you're right. Could you explain what this is doing?

Reply:
I actually had to fix this one, so thank you for pointing me to it.  It was reversed from how it should be.

In GNU, ./ls -aliL *:

```
test:
./ls: cannot access 'dangle': No such file or directory
total 0
72095191 drwxr-xr-x 11 rswinford staff 352 Dec 28 23:05 .
71925132 drwxr-xr-x 13 rswinford staff 416 Dec 29 13:19 ..
? l?????????    ?         ?       ?            ? dangle -> dne
72095193 -rw-r--r--  1 rswinford staff   0 Dec 28 22:55 test1
```

Whereas, in GNU, ./ls -alRiL:
```
./test:
./ls: cannot access 'dangle': No such file or directory
total 0
72095191 drwxr-xr-x 11 rswinford staff 352 Dec 28 23:05 .
71925132 drwxr-xr-x 13 rswinford staff 416 Dec 29 13:19 ..
       ? l?????????    ?         ?       ?            ? dangle -> dne
```

Reply:
Ah good catch! We'll keep that in then!